src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -810,4 +810,199 @@ public static void emitNodeMetrics(
           "Total time taken for writing {} metrics metricsdb: {}", tableName, mFinalT - mCurrT);
     }
   }
+
+  public static void emitShardStateMetric(
+          MetricsDB metricsDB, ShardStateMetricsSnapshot shardStateMetricsSnapshot) {
+    long mCurrT = System.currentTimeMillis();
+    Result<Record> shardStateMetrics = shardStateMetricsSnapshot.fetchAggregatedShardStateMetrics();
+
+    List<String> dims =
+            new ArrayList<String>() {
+              {
+                this.add(AllMetrics.ShardStateDimension.INDEX_NAME.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_ID.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_TYPE.toString());
+                this.add(AllMetrics.ShardStateDimension.NODE_NAME.toString());
+              }
+            };
+
+    emitActiveShardStateMetric(metricsDB, shardStateMetrics, dims);
+    emitInitializingShardStateMetric(metricsDB, shardStateMetrics, dims);
+    emitUnassignedShardStateMetric(metricsDB, shardStateMetrics, dims);
+
+    long mFinalT = System.currentTimeMillis();
+    LOG.debug(
+            "Total time taken for writing shard state event queue metrics metricsdb: {}", mFinalT - mCurrT);
+  }
+
+  public static void emitActiveShardStateMetric(MetricsDB metricsDB, Result<Record> res, List<String> dims) {
+    metricsDB.createMetric(
```

Comment:
I am thinking if we can save on bytes by only accounting for unassigned and initializing shards (leaving out the active shards). What remains are the active shards and the happy case.

Reply:
Published all shard states except Active